pkg/controller/jobframework/reconciler.go

```diff
@@ -439,6 +440,23 @@ func (r *JobReconciler) getParentWorkload(ctx context.Context, job GenericJob, o
 func (r *JobReconciler) ensureOneWorkload(ctx context.Context, job GenericJob, object client.Object) (*kueue.Workload, error) {
 	log := ctrl.LoggerFrom(ctx)
 
+	if usePrebuiltWorkload, prebuiltWorkloadName := prebuiltWorkload(job); usePrebuiltWorkload {
+		wl := &kueue.Workload{}
+		err := r.client.Get(ctx, types.NamespacedName{Name: prebuiltWorkloadName, Namespace: object.GetNamespace()}, wl)
+		if err != nil {
+			return nil, client.IgnoreNotFound(err)
+		}
+
+		if owns, err := r.ensurePrebuiltWorkloadOwnership(ctx, wl, object); !owns || err != nil {
```

Comment:
For consistency with the rest of the code, use an UnretriableError, instead of the extra boolean https://github.com/kubernetes-sigs/kueue/blob/main/pkg/controller/jobframework/errors.go